server/src/unifyfs_request_manager.c

```diff
@@ -1440,6 +1444,90 @@ static int process_unlink_rpc(reqmgr_thrd_t* reqmgr,
     return ret;
 }
 
+static int process_node_local_extents_get_rpc(reqmgr_thrd_t* reqmgr,
+                                              client_rpc_req_t* req)
+{
+    int ret = UNIFYFS_SUCCESS;
+    unifyfs_node_local_extents_get_in_t* in = req->input;
+    assert(in != NULL);
+    size_t num_req = in->num_req;
+    margo_free_input(req->handle, in);
+    free(in);
+    unifyfs_extent_t* in_extents = (unifyfs_extent_t*) req->bulk_buf;
+    struct extents_list out_list;
+    struct extents_list* out_list_cur = &out_list;
+    size_t total_chunks = 0;
+    bool add_node = false;
+    for (int i = 0; i < num_req; ++i) {
+        LOGDBG("getting node local extent for gfid=%d", in_extents[i].gfid);
+        unsigned int n_chunks = 0;
+        chunk_read_req_t* chunks = NULL;
+        int rc = unifyfs_invoke_find_extents_rpc(in_extents[i].gfid, 1,
+                                                 &in_extents[i],
+                                                 &n_chunks, &chunks);
+        if (!rc) {
+            if (add_node) {
+                out_list_cur->next = calloc(1, sizeof(struct extents_list));
+                out_list_cur = out_list_cur->next;
+            }
+            for (int j = 0; j < n_chunks; ++j) {
+                out_list_cur->value.file_pos = chunks[j].offset;
+                out_list_cur->value.log_pos = chunks[j].log_offset;
+                out_list_cur->value.length = chunks[j].nbytes;
+                out_list_cur->value.gfid = in_extents[j].gfid;
+                out_list_cur->value.log_app_id = chunks[j].log_app_id;
+                out_list_cur->value.log_client_id = chunks[j].log_client_id;
+                if (i < n_chunks - 1) {
+                    out_list_cur->next = calloc(1, sizeof(struct extents_list));
+                    out_list_cur = out_list_cur->next;
+                } else {
+                    out_list_cur->next = NULL;
+                }
+            }
+            add_node = true;
+            total_chunks += n_chunks;
+            free(chunks);
+        }
+    }
+    free(req->bulk_buf);
+    /* prepare the response for node local extents get */
+    unifyfs_node_local_extents_get_out_t out;
+    hg_return_t hret = 0;
+    out.ret = (int32_t) ret;
+    out.extent_count = total_chunks;
+    size_t extents_size = total_chunks*sizeof(unifyfs_client_index_t);
+    unifyfs_client_index_t* extents_buffer =
+            malloc(extents_size);
+    extents_list* cur = &out_list;
+    for (int i = 0; i < total_chunks; ++i) {
+        extents_buffer[i].file_pos = cur->value.file_pos;
+        extents_buffer[i].length = cur->value.length;
+        extents_buffer[i].gfid = cur->value.gfid;
```

Comment:
I think you are missing assignments to the log-related fields of the `unifyfs_client_index_t` here

Reply:
Done